core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -75,7 +79,12 @@ public TableDataMap(AbsoluteTableIdentifier identifier, String dataMapName,
     SegmentProperties segmentProperties;
     for (String segmentId : segmentIds) {
       List<Blocklet> pruneBlocklets = new ArrayList<>();
-      List<DataMap> dataMaps = dataMapFactory.getDataMaps(segmentId);
+      List<DataMap> dataMaps;
+      if (blockletDetailsFetcher instanceof DataMapFactory && filterExp == null) {
+        dataMaps = ((DataMapFactory)blockletDetailsFetcher).getDataMaps(segmentId);
```

Comment:
Please add correct generic assignment, it is hard to check whether the type is correct or not

Reply:
I have added here but placing generic assignment all places is not possible because we are deriving two types of datamaps through generic .